File: code_examples/5-Holding_Data/1-Arrays/Java/Main.java
/*
 * This program uses an array
 * .... this was linted by GitHub Actions
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    // this function uses an array

    int[] studentMarks = new int[5];
    int aSingleMark;

    // get length of array
    int arrayLength = studentMarks.length;

    // input
    Scanner scanner = new Scanner(System.in);
    for (int loop_counter = 0; loop_counter < arrayLength; loop_counter++) {
        System.out.print("Enter a mark (percentage): ");
        aSingleMark = scanner.nextInt();
        studentMarks[loop_counter] = aSingleMark;
    }
    System.out.println();
    System.out.println("Here are the 5 marks:");

    for (int loop_counter = 0; loop_counter < arrayLength; loop_counter++) {
        System.out.print(studentMarks[loop_counter] + "% ");
    }

    System.out.println();
    System.out.println("\nDone.");
  }
}

File: code_examples/5-Holding_Data/1-Arrays/CPP/main.cpp
// Copyright (c) 2019 St. Mother Teresa HS All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Dec 2019
// This program uses an array

#include <iostream>


int main() {
    // this function uses an array

    int studentMarks[5];
    int aSingleMark;

    // input
    for (int loop_counter = 0; loop_counter < 5; loop_counter++) {
        std::cout << "Enter a mark (percentage): ";
        std::cin >> aSingleMark;
        studentMarks[loop_counter] = aSingleMark;
    }
    std::cout << std::endl;

    std::cout << "Here are the 5 marks:" << std::endl;

    // get length of array
    int arrayLength = sizeof(studentMarks) / sizeof(studentMarks[0]);

    for (int loop_counter = 0; loop_counter < arrayLength; loop_counter++) {
        std::cout << studentMarks[loop_counter] << "% ";
    }

    std::cout << "" << std::endl;
    std::cout << "\nDone." << std::endl;
    return 0;
}
